datadog-ffe/src/rules_based/error.rs

```diff
@@ -1,53 +1,10 @@
 // Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
-use std::sync::Arc;
-
 use serde::{Deserialize, Serialize};
 
 use crate::rules_based::ufc::VariationType;
 
-/// Represents a result type for operations in the feature flagging SDK.
-///
-/// This type alias is used throughout the SDK to indicate the result of operations that may return
-/// errors specific to the feature flagging SDK.
-///
-/// This `Result` type is a standard Rust `Result` type where the error variant is defined by the
-/// ffe-specific [`Error`] enum.
-pub type Result<T> = std::result::Result<T, Error>;
-
-/// Enum representing possible errors that can occur in the feature flagging SDK.
```

Comment:
Just an FYI: removing this shouldn't impact anything from the FFI's perspective. The FFI converts all errors to a generic `ddog_Error` type using a `wrap_with_ffi_result` method

Within `wrap_with_ffi_result`, there's a `anyhow::Error → ddcommon_ffi::Error` conversion that flattens all error details into a string message

https://github.com/DataDog/libdatadog/blob/4828b758f7002266811697e1f70be2de9725c6b4/ddcommon-ffi/src/error.rs#L62-L68

And so there's no distinction of Rust-specific error types once they get to C

Reply:
Thank you for referring me to that PR, particularly the methods you have to map the native errors to OF errors are helpful: `_map_reason_to_openfeature` and `_map_error_code_to_openfeature`. We'll update Ruby to do the same mapping.

Moving `ResolutionDetails` into `datadog-ffe` doesn't sound like a bad idea. If the OF specs for python, ruby, and other SDKs never drift, that would be ideal. But even if one of them diverges, I imagine we could just augment the code for that specific SDK